crates/circuit/src/dag_circuit.rs

```diff
@@ -2551,25 +2553,71 @@ def _format(operand):
-        //     self._multi_graph.add_edge(node1_id, edge_child._node_id, edge)
-        todo!()
+        let node1 = node1.node.unwrap();
+        let node2 = node2.node.unwrap();
+
+        // Gather all wires connecting node1 and node2.
+        // This functionality was extracted from rustworkx's 'get_edge_data'
+        let wires: Vec<Wire> = self
+            .dag
+            .edges(node1)
+            .filter(|edge| edge.target() == node2)
+            .map(|edge| edge.weight().clone())
+            .collect();
+
+        if wires.is_empty() {
+            return Err(DAGCircuitError::new_err(
+                "Attempt to swap unconnected nodes",
+            ));
+        };
+
+        // Closure that finds the first parent/child node connected to a reference node by given wire
+        // and returns relevant edge information depending on the specified direction:
+        //  - Incoming -> parent -> outputs (parent_edge_id, parent_source_node_id)
+        //  - Outgoing -> child -> outputs (child_edge_id, child_target_node_id)
+        // This functionality was inspired in rustworkx's 'find_predecessors_by_edge' and 'find_successors_by_edge'.
+        // To understand name, see: https://en.wikipedia.org/wiki/Oyakodon
+        let find_first_oyako_by_wire =
+            |(node, direction, &ref wire): (NodeIndex, Direction, &Wire)| {
+                let mut oyakosan = Vec::new();
+                for edge in self.dag.edges_directed(node, direction) {
+                    if wire == edge.weight() {
+                        match direction {
+                            Incoming => oyakosan.push((edge.id(), edge.source())),
+                            Outgoing => oyakosan.push((edge.id(), edge.target())),
+                        };
+                        break;
+                    }
+                }
+                oyakosan[0]
+            };
+
+        // Vector that contains a tuple of (wire, edge_info, parent_info, child_info) per wire in wires
+        let relevant_edges = wires
+            .iter()
+            .rev()
+            .map(|wire| {
+                (
+                    wire,
+                    find_first_oyako_by_wire((node1, Outgoing, wire)),
+                    find_first_oyako_by_wire((node1, Incoming, wire)),
+                    find_first_oyako_by_wire((node2, Outgoing, wire)),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        // Iterate over relevant edges and modify self.dag
+        for (wire, (edge_id, _), (parent_id, parent_source), (child_id, child_target)) in
+            relevant_edges
+        {
+            self.dag.remove_edge(parent_id);
+            self.dag.add_edge(parent_source, node2, *wire);
+            self.dag.remove_edge(edge_id);
+            self.dag.add_edge(node2, node1, *wire);
+            self.dag.remove_edge(child_id);
+            self.dag.add_edge(node1, child_target, *wire);
+        }
+        Ok(())
```

Comment:
```suggestion
        for (wire, (node1_to_node2, _), (parent_to_node1, parent), (node2_to_child, child)) in
            relevant_edges
        {
            self.dag.remove_edge(parent_to_node1);
            self.dag.add_edge(parent, node2, *wire);
            self.dag.remove_edge(node1_to_node2);
            self.dag.add_edge(node2, node1, *wire);
            self.dag.remove_edge(node2_to_child);
            self.dag.add_edge(node1, child, *wire);
        }
        Ok(())
```

I think it might help readability quite a bit to rename these slightly so that the edge removals have symmetry to what is being added in as their replacements.